compiler/rustc_mir_build/src/builder/matches/test.rs

```diff
@@ -575,8 +577,8 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     None
                 } else {
                     fully_matched = true;
-                    let bits = value.eval_bits(self.tcx, self.typing_env());
-                    Some(TestBranch::Constant(value, bits))
+                    let bits = value.try_to_scalar_int().unwrap().to_bits_unchecked();
+                    Some(TestBranch::Constant(value.valtree, bits))
```

Comment:
Something very odd is happening here... this seems to be the only place that constructs a `TestBranch::Constant`, so `bits` here is *always*` the result of calling `unwrap_leaf().to_bits_unchecked()` on the valtree. So, uh, why do we store both the valtree and its bits?